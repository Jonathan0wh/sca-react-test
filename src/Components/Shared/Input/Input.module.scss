@import "styles/colors";

.input_wrapper {
  display: flex;
  flex-direction: column;
  justify-content: left;

  @media (min-width: 768px) {
    margin-right: 36px;
  }
}

.label_text {
  color: $whitegrey;
  font-weight: bold;
}

.input {
  height: 40px;
  margin-top: 10px;
  margin-bottom: 10px;
}

.asterisk {
  color: $red;
}
