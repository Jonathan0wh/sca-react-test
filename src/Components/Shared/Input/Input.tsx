import React from 'react';

import styles from './Input.module.scss';

interface InputProps {
  labelName: string;
  inputType: string;
  required: boolean;
  dataField: string;
  handleInputChange: Function;
}

const Input = ({
  labelName,
  inputType,
  required,
  dataField,
  handleInputChange
}: InputProps) => (
  <div className={styles.input_wrapper}>
    <label htmlFor={`input-${labelName}`} className={styles.label_text}>
      {labelName} {required && <span className={styles.asterisk}>*</span>}
    </label>
    <input
      className={styles.input}
      id={`input-${labelName}`}
      type={inputType}
      required={required}
      onChange={event => handleInputChange(dataField, event.target.value)}
    />
  </div>
);

export default Input;
