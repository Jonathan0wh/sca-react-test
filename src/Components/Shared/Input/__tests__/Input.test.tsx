import React from 'react';
import { shallow } from 'enzyme';

import Input from '../Input';

describe('<Input />', () => {
  const handleInputChangeFn = jest.fn();

  it('render Input as normal (not required)', () => {
    const wrapper = shallow(
      <Input
        labelName="test"
        inputType="text"
        dataField="test"
        handleInputChange={handleInputChangeFn}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('render Input as required', () => {
    const wrapper = shallow(
      <Input
        labelName="test"
        inputType="text"
        required
        dataField="test"
        handleInputChange={handleInputChangeFn}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
