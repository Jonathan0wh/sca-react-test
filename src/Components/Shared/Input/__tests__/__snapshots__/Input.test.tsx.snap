// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> render Input as normal (not required) 1`] = `
<div
  className="input_wrapper"
>
  <label
    className="label_text"
    htmlFor="input-test"
  >
    test
     
  </label>
  <input
    className="input"
    id="input-test"
    onChange={[Function]}
    type="text"
  />
</div>
`;

exports[`<Input /> render Input as required 1`] = `
<div
  className="input_wrapper"
>
  <label
    className="label_text"
    htmlFor="input-test"
  >
    test
     
    <span
      className="asterisk"
    >
      *
    </span>
  </label>
  <input
    className="input"
    id="input-test"
    onChange={[Function]}
    required={true}
    type="text"
  />
</div>
`;
